index.js

```diff
@@ -32,6 +32,10 @@ module.exports = fp(function from (fastify, opts, next) {
     const rewriteRequestHeaders = opts.rewriteRequestHeaders || requestHeadersNoOp
     const getUpstream = opts.getUpstream || upstreamNoOp
     const onError = opts.onError || onErrorDefault
+    const contentTypesToEncode = new Set([
+      'application/json',
+      ...(opts.contentTypesToEncode || [])
+    ])
```

Comment:
This set and the relative option should be moved one level up as creating a new set every time is relatively expensive.